cmd/xl-storage.go

```diff
@@ -1784,10 +1784,10 @@ func (s *xlStorage) CheckFile(ctx context.Context, volume string, path string) e
 	}
 
 	// Stat a volume entry.
-	_, err = os.Stat(volumeDir)
+	_, err = os.Stat(pathJoin(volumeDir, path))
 	if err != nil {
 		if osIsNotExist(err) {
-			return errVolumeNotFound
+			return errPathNotFound
 		}
 		return err
 	}
```

Comment:
Changing might get confusing for top layer, since this is a new error it would be harder to handle at all layers. We shouldn't introduce new err's at storage layer. 

You can still use `errFileNotFound` here @vadmeste

Reply:
> Changing might get confusing for top layer, since this is a new error it would be harder to handle at all layers. We shouldn't introduce new err's at storage layer.
> 
> You can still use `errFileNotFound` here @vadmeste

CheckFile() is only called from one location, which is er.isObject()

Besides errFileNotFound means a prefix can exist but it is not an object, and I really need to return the prefix does not exist as well so er.isObject() can return quickly.

Reply:
No you are looking for the path and path here is a file. We shouldn't introduce new error codes specific to single storage calls.

You can use errFileNotFound without any issues here.